src/hooks/useValidateTransactionMemo.ts

```diff
@@ -0,0 +1,136 @@
+import { TransactionBuilder } from "@stellar/stellar-sdk";
+import {
+  mapNetworkToNetworkDetails,
+  STORAGE_KEYS,
+  TRANSACTION_WARNING,
+} from "config/constants";
+import { MemoRequiredAccountsApiResponse } from "config/types";
+import { useAuthenticationStore } from "ducks/auth";
+import { usePreferencesStore } from "ducks/preferences";
+import { useTransactionBuilderStore } from "ducks/transactionBuilder";
+import { useTransactionSettingsStore } from "ducks/transactionSettings";
+import { cachedFetch } from "helpers/cachedFetch";
+import { isMainnet } from "helpers/networks";
+import { getApiStellarExpertIsMemoRequiredListUrl } from "helpers/stellarExpert";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { stellarSdkServer } from "services/stellar";
+
+export const useValidateTransactionMemo = () => {
+  const { network } = useAuthenticationStore();
+  const { transactionXDR } = useTransactionBuilderStore();
+  const { isMemoValidationEnabled } = usePreferencesStore();
+  const { transactionMemo } = useTransactionSettingsStore();
+  const [isValidatingMemo, setIsValidatingMemo] = useState(false);
+  const networkDetails = useMemo(
+    () => mapNetworkToNetworkDetails(network),
+    [network],
+  );
+
+  const shouldValidateMemo = useMemo(
+    () => isMemoValidationEnabled && isMainnet(network),
+    [isMemoValidationEnabled, network],
+  );
+  const [isMemoMissing, setIsMemoMissing] = useState(shouldValidateMemo);
+
+  const checkMemoRequiredFromCache = useCallback(
+    async (txXDR: string, networkType: string): Promise<boolean> => {
```

Comment:
```suggestion
    async (transactionXDR: string, networkType: string): Promise<boolean> => {
```

let's give preference to using `transaction` instead of `tx` whenever possible so we keep the codebase consistency

Reply:
hummm ok I just saw we already have a `transactionXDR` from `useTransactionBuilderStore` so we'd need to use a different name here